src/api.ts

```diff
@@ -59,6 +59,126 @@ function addSkill(devId: number, newSkill: string): boolean {
   return false;
 }
 
+function updateSkill(
+  devId: number,
+  oldSkill: string,
+  newSkill: string,
+): boolean {
+  const developer = developers.find(
+    ({ id: existingDevId }) => existingDevId === devId,
+  );
+
+  if (!developer) {
+    return false;
+  }
+
+  const { skills } = developer;
+
+  if (
+    !skills.some((existingSkill) => existingSkill === newSkill) &&
+    skills.some((existingSkill) => existingSkill === oldSkill)
+  ) {
+    developer.skills[
+      developer.skills.findIndex((existingSkill) => existingSkill === oldSkill)
+    ] = newSkill;
+    return true;
+  }
```

Comment:
Skills are already destructured, then why using developer.skills. findIndex instead of skills. findIndex

Reply:
Yeah, my mistake, I forgot to update those lines